src/lib.rs

```diff
@@ -1311,6 +1312,118 @@ where
     Some(Ratio::new(n1, d1))
 }
 
+impl<T: Clone + Integer + Signed + ToPrimitive + ToBigInt> Ratio<T> {
+    /// Converts the ratio to an `f64`.
+    ///
+    /// The value is guaranteed to be the closest `f64` approximating the ratio. This always
+    /// returns a value, unless the conversion to BigInt for the numerator or denominator fails.
+    ///
+    /// ```
+    /// use num_rational::Rational64;
+    /// assert_eq!(0.5f64, Rational64::new(1, 2).to_f64().unwrap());
+    /// ```
+    pub fn to_f64(&self) -> Option<f64> {
+        assert_eq!(
+            std::f64::RADIX,
+            2,
+            "only floating point implementations with radix 2 are currently supported"
+        );
+        assert!(
+            std::mem::size_of::<isize>() >= 4,
+            "platforms 16 bits and under are not supported"
```

Comment:
Is this check necessary? (there is a [16-bit architecture](https://forge.rust-lang.org/platform-support.html#tier-3) with tier 3 Rust support). And I don't know anything about 16-bit MSP430, but it might support floats.